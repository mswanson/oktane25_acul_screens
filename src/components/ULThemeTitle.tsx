import type { HTMLAttributes, ReactNode } from "react";

import { cn } from "@/lib/utils";

export interface ULThemeTitleProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * The title of the screen.
   */
  children: ReactNode;
  /**
   * Optional class names for additional styling or overriding default styles.
   */
  className?: string;
}

const ULThemeTitle = ({ children, className, ...rest }: ULThemeTitleProps) => {
  const themedStyles =
    "mt-6 mb-4 text-header justify-text-header text-(length:--ul-theme-font-title-size) font-title";

  return (
    <h1 className={cn(themedStyles, className)} {...rest}>
      {children}
    </h1>
  );
};

export default ULThemeTitle;
// Test comment
// Another test comment
