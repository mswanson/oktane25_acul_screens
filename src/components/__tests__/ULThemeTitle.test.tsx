import { render } from "@testing-library/react";

import ULThemeTitle from "@/components/ULThemeTitle";

describe("ULThemeTitle Component", () => {
  // Snapshot Tests
  it("matches snapshot with default props", () => {
    const { container } = render(<ULThemeTitle>Test Title</ULThemeTitle>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with custom className", () => {
    const { container } = render(
      <ULThemeTitle className="custom-class">Test Title</ULThemeTitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with additional props", () => {
    const { container } = render(
      <ULThemeTitle id="test-id" data-testid="title">
        Test Title
      </ULThemeTitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with multiple children", () => {
    const { container } = render(
      <ULThemeTitle>
        <span>Child 1</span>
        <span>Child 2</span>
      </ULThemeTitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Functional Tests
  it("renders children correctly", () => {
    const { getByText } = render(<ULThemeTitle>Test Title</ULThemeTitle>);
    expect(getByText("Test Title")).toBeInTheDocument();
  });

  it("applies default themed styles", () => {
    const { container } = render(<ULThemeTitle>Test Title</ULThemeTitle>);
    expect(container.firstChild).toHaveClass(
      "mt-6 mb-4 text-header justify-text-header text-(length:--ul-theme-font-title-size) font-title"
    );
  });

  it("applies custom className", () => {
    const { container } = render(
      <ULThemeTitle className="custom-class">Test Title</ULThemeTitle>
    );
    expect(container.firstChild).toHaveClass("custom-class");
  });

  it("passes additional props to the root element", () => {
    const { container } = render(
      <ULThemeTitle id="test-id" data-testid="title">
        Test Title
      </ULThemeTitle>
    );
    expect(container.firstChild).toHaveAttribute("id", "test-id");
    expect(container.firstChild).toHaveAttribute("data-testid", "title");
  });
});
