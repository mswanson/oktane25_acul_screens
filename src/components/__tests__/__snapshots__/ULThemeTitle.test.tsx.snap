// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemeTitle Component matches snapshot with additional props 1`] = `
<h1
  class="mt-6 mb-4 text-header justify-text-header text-(length:--ul-theme-font-title-size) font-title"
  data-testid="title"
  id="test-id"
>
  Test Title
</h1>
`;

exports[`ULThemeTitle Component matches snapshot with custom className 1`] = `
<h1
  class="mt-6 mb-4 text-header justify-text-header text-(length:--ul-theme-font-title-size) font-title custom-class"
>
  Test Title
</h1>
`;

exports[`ULThemeTitle Component matches snapshot with default props 1`] = `
<h1
  class="mt-6 mb-4 text-header justify-text-header text-(length:--ul-theme-font-title-size) font-title"
>
  Test Title
</h1>
`;

exports[`ULThemeTitle Component matches snapshot with multiple children 1`] = `
<h1
  class="mt-6 mb-4 text-header justify-text-header text-(length:--ul-theme-font-title-size) font-title"
>
  <span>
    Child 1
  </span>
  <span>
    Child 2
  </span>
</h1>
`;
